File: check_runs.js
import https from 'https';

const url = 'https://api.github.com/repos/gowtham5m11/leaderledger/actions/runs?per_page=20';
const options = {
  headers: {
    'User-Agent': 'Mozilla/5.0'
  }
};

https.get(url, options, (res) => {
  let data = '';
  res.on('data', (chunk) => {
    data += chunk;
  });
  res.on('end', () => {
    try {
      const json = JSON.parse(data);
      if (!json.workflow_runs) {
        console.log("No runs found in response:", json);
        return;
      }
      for (const run of json.workflow_runs) {
        console.log(`ID: ${run.id} | Name: ${run.name} | Event: ${run.event} | Status: ${run.status} | Conclusion: ${run.conclusion} | Created: ${run.created_at}`);
      }
    } catch (err) {
      console.error("Error parsing JSON:", err);
    }
  });
}).on('error', (err) => {
  console.error("Error fetching runs:", err);
});

File: check_runs.py
import urllib.request
import json

url = "https://api.github.com/repos/gowtham5m11/leaderledger/actions/runs?per_page=20"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
        for run in data.get('workflow_runs', []):
            print(f"ID: {run['id']} | Name: {run['name']} | Event: {run['event']} | Status: {run['status']} | Conclusion: {run['conclusion']} | Created: {run['created_at']}")
except Exception as e:
    print(f"Error: {e}")
